Result/Result.swift

```diff
@@ -31,7 +31,10 @@ public enum Result<T, Error: Swift.Error>: ResultProtocol, CustomStringConvertib
 	public init(attempt f: () throws -> T) {
 		do {
 			self = .success(try f())
-		} catch {
+		} catch var error {
+			if Error.self == AnyError.self {
```

Comment:
This should probably only be if `error` _isn't_ an `AnyError`? (Can you add a test for that?)